### PR TITLE
Export Functionality Added

### PR DESCRIPTION
**Pull request by dmrt2002**
export class added with functions to convert the table data to csv and pdf downloadable files respectively

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +31,9 @@
 		"vue": "^3.2.13",
 		"vue-router": "^4.0.3",
 		"vue-sfc-rollup": "^4.0.5",
+		"vue3-draggable": "^2.0.9",
```
Comment (TheCelebrimbor) [inline package.json]: Can we remove unused packages ?

File: src/components/tuPagination/tuPagination.vue [hunk 2]
```diff
@@ -100,7 +100,8 @@ export default defineComponent({
 		loadingItems: { default: () => [], type: Array },
 		length: { default: 1, type: Number },
 		max: { default: 9, type: Number },
-		dottedNumber: { default: 5, type: Number }
+		dottedNumber: { default: 5, type: Number },
+		persistentId: { default: 1, type: Number }
```
Comment (TheCelebrimbor) [inline src/components/tuPagination/tuPagination.vue]: Change id type to string

File: src/components/tuTable/tuTable.vue [hunk 3]
```diff
@@ -61,10 +61,16 @@
 							minWidth: header.minWidth,
 							maxWidth: header.maxWidth
 						}"
+                        draggable="true"
```
Comment (TheCelebrimbor) [inline src/components/tuTable/tuTable.vue]: can we make this draggable dynamic ? Make it true only if a prop is true ?

something like 
```js
    :draggable="isDraggable"
    
    ..
    ..
    props:  {
        isDraggable: Boolean
    }
```

**Review by TheCelebrimbor — COMMENTED** [review]
Make sure all eslint code rules are enforced/fixed

File: src/components/tuTable/tuTableStore.ts [hunk 4]
```diff
@@ -149,13 +164,19 @@ export class TuTableStore {
...
-						else if (fieldValue !== undefined)
-							return String(fieldValue).toLowerCase().includes(String(filterValue.value).toLowerCase());
-						else
-							return true;
+						const fieldValue = this.getField(
+							value.rowData,
+							filterValue.field
+						);
+						if (filterValue.value === "") return true;
+						else if (fieldValue !== undefined) {
```
Comment (rvndbalaji) [inline src/components/tuTable/tuTableStore.ts]: You dont need `else` if the first statement is doing a return, just use `if`

File: src/components/tuTable/tuTableStore.ts [hunk 5]
```diff
@@ -345,10 +389,59 @@ export class TuTableStore {
...
+			const getNextHeader = (i: number) => {
+				const header = persistentColumns.columns[i].header;
+				let requiredHeader: TuHeaderDefn;
+				for (let j = 0; j < this.table.headers.length; j++) {
+					if (this.table.headers[j].field === header) {
+						requiredHeader = this.table.headers[j];
+						break;
+					}
+				}
+				return requiredHeader;
```
Comment (rvndbalaji) [inline src/components/tuTable/tuTableStore.ts]: Move the `return` statement and replace the `break` with it.
You dont need to use break and then return. Directly return

File: src/components/tuTable/tuTableStore.ts [hunk 6]
```diff
@@ -345,10 +389,59 @@ export class TuTableStore {
...
+					indexOneFound = true;
+				}
+			}
+			if (!indexTwoFound) {
+				if (this.table.headers[i].index === indexTwo) {
+					indexTwo = i;
+					indexTwoFound = true;
+				}
+			}
+		}
```
Comment (rvndbalaji) [inline src/components/tuTable/tuTableStore.ts:434]: At the end of loop, inside, do `if(indexOneFound && indexTwoFound) break;`. This way you stop looping once both flags are found

**Review by rvndbalaji — CHANGES_REQUESTED** [review]
Please address the changes

File: package.json (hunk 1 above)
Comment (dmrt2002) [reply package.json]: Okay

File: src/components/tuPagination/tuPagination.vue (hunk 2 above)
Comment (dmrt2002) [reply src/components/tuPagination/tuPagination.vue]: Yeah already done

File: src/components/tuTable/tuTable.vue (hunk 3 above)
Comment (dmrt2002) [reply src/components/tuTable/tuTable.vue]: Okay

File: src/components/tuTable/tuTableStore.ts (hunk 4 above)
Comment (dmrt2002) [reply src/components/tuTable/tuTableStore.ts]: Okay

File: src/components/tuTable/tuTableStore.ts (hunk 6 above)
Comment (dmrt2002) [reply src/components/tuTable/tuTableStore.ts:434]: Okay will do that

**Pull request closed without merging**